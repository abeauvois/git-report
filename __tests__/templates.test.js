const c = require("chalk");
const { monthTemplate, dayTemplate, htmlTemplate, consoleTemplate } = require("../src/templates");

let received, expected, fromCSVfile;

describe("templates", () => {
  beforeEach(() => {
    fromCSVfile = [
      9,
      [
        { day: 1, messages: "m1" },
        { day: 2, messages: "m2" },
      ],
    ];
    received = null;
    expected = null;
  });
  test("R1: monthTemplate: month in console", () => {
    received = monthTemplate(fromCSVfile[0], "console");
    expect(received).toMatchSnapshot();
  });

  test("R2: monthTemplate: month in html", () => {
    received = monthTemplate(fromCSVfile[0]);
    expect(received).toMatchSnapshot();
  });

  test("R3: dayTemplate: day in console", () => {
    const days = fromCSVfile[1];

    received = dayTemplate(days[0], "console");
    expect(received).toMatchSnapshot();
  });

  test("R4: dayTemplate: day in html", () => {
    const days = fromCSVfile[1];

    received = dayTemplate(days[0]);
    expect(received).toMatchSnapshot();
  });

  test("R5: htmlTemplate", () => {
    received = htmlTemplate(fromCSVfile);
    expect(received).toMatchSnapshot();
  });

  test("R5: consoleTemplate", () => {
    received = consoleTemplate(fromCSVfile);
    expect(received).toMatchSnapshot();
  });
});
