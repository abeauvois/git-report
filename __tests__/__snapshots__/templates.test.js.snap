// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`templates R1: monthTemplate: month in console 1`] = `"[34mNov[39m"`;

exports[`templates R2: monthTemplate: month in html 1`] = `"<strong>9, Nov</strong>"`;

exports[`templates R3: dayTemplate: day in console 1`] = `"[33m1[39m"`;

exports[`templates R4: dayTemplate: day in html 1`] = `"<strong>1</strong>"`;

exports[`templates R5: consoleTemplate 1`] = `[Function]`;

exports[`templates R5: htmlTemplate 1`] = `"Month <strong>9, Nov</strong><br><br>Day <strong>1</strong>: m1<br>Day <strong>2</strong>: m2<br><br>"`;
